src/parser/expression/constant_value_expression.cpp

```diff
@@ -250,6 +251,41 @@ std::string ConstantValueExpression::ToString() const {
   }
 }
 
+ConstantValueExpression ConstantValueExpression::FromString(const std::string &val, type::TypeId type_id) {
+  switch (type_id) {
+    case type::TypeId::BOOLEAN: {
+      if (val.empty()) return ConstantValueExpression(type_id, execution::sql::BoolVal::Null());
+      return ConstantValueExpression(type_id, execution::sql::BoolVal(std::stoi(val) != 0));
+    }
+    case type::TypeId::TINYINT:
+    case type::TypeId::SMALLINT:
+    case type::TypeId::INTEGER:
+    case type::TypeId::BIGINT: {
+      if (val.empty()) return ConstantValueExpression(type_id, execution::sql::Integer::Null());
+      return ConstantValueExpression(type_id, execution::sql::Integer(std::stoll(val)));
+    }
+    case type::TypeId::DECIMAL: {
+      if (val.empty()) return ConstantValueExpression(type_id, execution::sql::Real::Null());
+      return ConstantValueExpression(type_id, execution::sql::Real(std::stod(val)));
+    }
+    case type::TypeId::TIMESTAMP: {
+      if (val.empty()) return ConstantValueExpression(type_id, execution::sql::TimestampVal::Null());
+      return ConstantValueExpression(type_id, execution::sql::TimestampVal(execution::sql::Timestamp::FromString(val)));
+    }
+    case type::TypeId::DATE: {
+      if (val.empty()) return ConstantValueExpression(type_id, execution::sql::DateVal::Null());
+      return ConstantValueExpression(type_id, execution::sql::DateVal(execution::sql::Date::FromString(val)));
+    }
+    case type::TypeId::VARCHAR:
+    case type::TypeId::VARBINARY: {
+      auto string_val = execution::sql::ValueUtil::CreateStringVal(val);
```

Comment:
We seems to be missing NULL for variable length types.

Reply:
Updated using the general constructor.